### ISSUE TITLE
Can't handle out of api quota condition.

### ISSUE THREAD

**Issue by yubbie** [jay0lee/GAM#253]
The issue tracker is for reporting product deficiencies. How do I questions should be posted to the discussion forum at https://groups.google.com/group/google-apps-manager. When in doubt, start at the discussion forum and return here only when instructed to do so.

Please confirm the following:
* I have upgraded to the latest GAM release from http://git.io/gamreleases and I still have this issue.
No, 3.65 still.  Need things stable for a few days so I can't go to 3.7 yet.
* I am typing the command as described in the GAM Wiki at https://github.com/jay0lee/GAM/wiki
yes
* I am using a Google Apps for Work, Unlimited or EDU account (Google Apps Standard / Free is not supported)
Yes 

Full steps to reproduce the issue:
1.  Try to do a gam update group when you're out of admin sdk quota.  I'm at 156,000 of my 150,000 requests/day right now.  (I've asked google for more)
 

Expected outcome (what are you trying to do?):
Expected to get a graceful error message telling me I'm out of luck because out of quota is out of quota.

Actual outcome (what errors or bad behavior do you see instead?):

Big dump...

/usr/bin/python /home/jarrell/gam3/gam.py update group class-engl-3704-13899-201701-g@vt.edu sync member file class-engl-3704-13899-201701-g.sub;/usr/bin/python /home/jarrell/gam3/gam.py update group class-engl-3704-13899-201701-g@vt.edu sync owner file class-engl-3704-13899-201701-g.owner
No handlers could be found for logger "googleapiclient.discovery"
Traceback (most recent call last):
  File "/home/jarrell/gam3/gam.py", line 9213, in <module>
    doUpdateGroup()
  File "/home/jarrell/gam3/gam.py", line 6019, in doUpdateGroup
    cd = buildGAPIObject(u'directory')
  File "/home/jarrell/gam3/gam.py", line 813, in buildGAPIObject
    service = googleapiclient.discovery.build(api, version, http=http, cache_discovery=False)
  File "/home/jarrell/gam3/oauth2client/util.py", line 137, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/jarrell/gam3/googleapiclient/discovery.py", line 203, in build
    cache)
  File "/home/jarrell/gam3/googleapiclient/discovery.py", line 263, in _retrieve_discovery_doc
    raise InvalidJsonError()
googleapiclient.errors.InvalidJsonError
No handlers could be found for logger "googleapiclient.discovery"
Traceback (most recent call last):
  File "/home/jarrell/gam3/gam.py", line 9213, in <module>
    doUpdateGroup()
  File "/home/jarrell/gam3/gam.py", line 6019, in doUpdateGroup
    cd = buildGAPIObject(u'directory')
  File "/home/jarrell/gam3/gam.py", line 813, in buildGAPIObject
    service = googleapiclient.discovery.build(api, version, http=http, cache_discovery=False)
  File "/home/jarrell/gam3/oauth2client/util.py", line 137, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/jarrell/gam3/googleapiclient/discovery.py", line 203, in build
    cache)
  File "/home/jarrell/gam3/googleapiclient/discovery.py", line 263, in _retrieve_discovery_doc
    raise InvalidJsonError()
googleapiclient.errors.InvalidJsonError

**Comment by jay0lee** [comment]
This looks like a corrupt cache. What error do you get if you create a file called nocache.txt in the same path as gam.py?

**Comment by yubbie** [comment]
Google Apps Domain: vt.edu
No handlers could be found for logger "googleapiclient.discovery"
Traceback (most recent call last):
  File "/home/jarrell/gam3/gam", line 9276, in <module>
    doGetDomainInfo()
  File "/home/jarrell/gam3/gam", line 1421, in doGetDomainInfo
    doGetInstanceInfo()
  File "/home/jarrell/gam3/gam", line 7244, in doGetInstanceInfo
    cd = buildGAPIObject(u'directory')
  File "/home/jarrell/gam3/gam", line 813, in buildGAPIObject
    service = googleapiclient.discovery.build(api, version, http=http, cache_discovery=False)
  File "/home/jarrell/gam3/oauth2client/util.py", line 137, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/jarrell/gam3/googleapiclient/discovery.py", line 203, in build
    cache)
  File "/home/jarrell/gam3/googleapiclient/discovery.py", line 263, in _retrieve_discovery_doc
    raise InvalidJsonError()
googleapiclient.errors.InvalidJsonError


Also does it with 3.7 - just set up a parallel installation.

**Comment by yubbie** [comment]
That was from a "gam info domain", btw.

**Comment by yubbie** [comment]
Then again.  I think I crossed the streams.  3.7 was working last night after midnight when my quota reset, but 3.65 still wasn't, but 3.7 was the one with the nocache.txt file.  I had cloned the gamcache when I cloned the 3.65 tree to apply the 3.7 files.  I deleted gamcache, and now they're both working.  But it still seems to have triggered right about the time I ran out of adminsdk quota.  Which, since google bumped me to 400K, hopefuly won't be a problem for a long, long, time.
